### PR TITLE
Pull Request for Issue1451: updating mono region control.

### PR DESCRIPTION
**Pull request by helfrij**
- Updated the SSRL mono classes: variable names, macro names, using AMActionSupport to create move actions. Replaced the CLSMAXvMotors in the Side and Main mono classes used in the crystal change with BioXASMAXvMotors.
- Modified the 'move to limit' action to be able to handle when the motor reports a move done before the limit control reports that the motor is at a limit.
- Modified the CLSControlEditor to include a flag for whether or not the editor is the one that initiated a control's current move.

### CODE REVIEW

**Comment by helfrij** [comment]
Hey @iainworkman, would you review?

File: clsCommon.pri [hunk 1]
```diff
@@ -14,7 +14,6 @@ HEADERS *=\
 	source/beamline/CLS/CLSValves.h \
 	source/beamline/CLS/CLSShutters.h \
 	source/beamline/CLS/CLSMirrorMaskState.h \
-	source/beamline/CLS/CLSSSRLMonochromatorMaskState.h \
```
Comment (iainworkman) [inline clsCommon.pri]: Is this something which got moved up to a CLS class in the recent refactor, but didn't really make sense as a CLS class?

File: source/ui/BioXAS/BioXASSSRLMonochromatorRegionControlEditor.cpp [hunk 2]
```diff
@@ -14,22 +14,21 @@ BioXASSSRLMonochromatorRegionControlEditor::~BioXASSSRLMonochromatorRegionContro
...
 }
 
-void BioXASSSRLMonochromatorRegionControlEditor::onRegionControlMoveStarted()
+void BioXASSSRLMonochromatorRegionControlEditor::showMovingView()
 {
-	if (control_ && control_->isConnected()) {
+	if (control_ && initiatedCurrentMove()) {
 		BioXASSSRLMonochromatorRegionControlMovingView *movingView = new BioXASSSRLMonochromatorRegionControlMovingView(qobject_cast<BioXASSSRLMonochromatorRegionControl*>(control_), this);
-		movingView->setWindowModality(Qt::WindowModal);
+		//movingView->setWindowModality(Qt::WindowModal);
```
Comment (iainworkman) [inline source/ui/BioXAS/BioXASSSRLMonochromatorRegionControlEditor.cpp]: Can this comment be removed?

**Comment by iainworkman** [comment]
Couple of very minor questions/comments.
Otherwise looks good to me.
Time Estimate: 25 minutes.

File: clsCommon.pri (hunk 1 above)
Comment (helfrij) [inline clsCommon.pri]: Yes, I think that's the case. The class isn't used anywhere else, and the rest of the SSRL mono classes are BioXAS-specific. If we'd like to move it up to the CLS level eventually, I'd prefer we move everything up together.

File: source/ui/BioXAS/BioXASSSRLMonochromatorRegionControlEditor.cpp (hunk 2 above)
Comment (helfrij) [inline source/ui/BioXAS/BioXASSSRLMonochromatorRegionControlEditor.cpp]: Yup, I'll remove.

File: source/beamline/BioXAS/BioXASMainMonochromator.cpp [hunk 3]
```diff
@@ -1,36 +1,87 @@
...
-	setRegionBStatus(new AMReadOnlyPVControl("RegionBStatus", "BL1607-5-I21:Mono:Region:B", this));
-	setVertical(new CLSMAXvMotor(QString("SMTR1607-5-I21-13 VERTICAL"), QString("SMTR1607-5-I21-13"), QString("SMTR1607-5-I21-13 VERTICAL"), true, 0.05, 2.0, this));
-	setLateral(new CLSMAXvMotor(QString("SMTR1607-5-I21-14 LATERAL"), QString("SMTR1607-5-I21-14"), QString("SMTR1607-5-I21-14 LATERAL"), true, 0.05, 2.0, this));
+	// Paddle controls.
+
+	BioXASMAXvMotor *paddle = new BioXASMAXvMotor(QString("SMTR1607-5-I21-11 PHOSPHOR PADDLE"), QString("SMTR1607-5-I21-11"), QString("SMTR1607-5-I21-11 PHOSPHOR PADDLE"), false, 0.1, 2.0, this);
+	paddle->setLimitSetpoint(CLSMAXvMotor::LimitCCW, BIOXASMAINMONOCHROMATOR_PADDLE_CCW_DESTINATION);
+	setPaddle(paddle);
+
+        setPaddleStatus(new AMReadOnlyPVControl(QString("%1PaddleStatus").arg(name), QString("BL1607-5-I21:Mono:PaddleExtracted"), this));
```
Comment (dretrex) [inline source/beamline/BioXAS/BioXASMainMonochromator.cpp]: Weird spacing here.

**Comment by dretrex** [comment]
This looks good!  Nice work!

Time Estimate: 15 minutes

**Pull request merged**